README.md

```diff
@@ -1,28 +1,182 @@
 # Velveteen [![Build Status](https://travis-ci.com/thoughtbot/velveteen.svg?branch=master)](https://travis-ci.com/thoughtbot/velveteen)
 
-🚧 A work in-progress 🚧
-
 Transform your background jobs into a real data pipeline with Velveteen.
 
-## Install
+## We've got a pipeline problem
+
+Background jobs are an inevitable addition for many Rails applications. They
+typically start with a need to send email, then quickly grow into a collection
+of interdependent data processing jobs that are yearning to be a pipeline.
+Velveteen aims to insert itself near the beginning, providing support to build
+out the pipeline in a maintainable way.
+
+## 🚧 A work in-progress 🚧
+
+Prior to version 1.0, minor version updates are susceptible to breaking changes.
+
+## Getting started
+
+Velveteen requires Ruby `>= 2.5` and access to RabbitMQ.
 
 Add the following line to Gemfile:
 
 ```ruby
 gem "velveteen"
 ```
 
-and run `bundle install` from your shell.
+Run the bundle command to install it.
+
+Define a worker:
+
+```ruby
+# do_something.rb
+require "velveteen"
+
+Velveteen::Config.exchange_name = "velveteen_development"
+
+class DoSomething < Velveteen::Worker
+  self.routing_key = "something.do"
+
+  def perform
+    # do something
+  end
+end
+```
+
+Run it:
+
+```shell
+velveteen work do_something.rb DoSomething
+```
+
+Within the worker, you have access to the following:
+
+* `message` - A representation of the Rabbit message, which responds to:
+  * `body` - the raw message body
+  * `data` - the parsed JSON representation of the body
+  * `delivery_info` - the delivery info from Rabbit
+  * `headers` - the headers from Rabbit, from `properties.headers`
+  * `properties` - the properties Rabbit
+* `publish(payload, [options])` - Publishes the message to the exchange
+
+  The headers from the message the worker is consuming will be automatically
+  included when publishing new messages.
+
+## Pipeline design
+
+In general, try to follow the best practices laid out in thoughtbot's [data
+guide].
+
+[data guide]: https://github.com/thoughtbot/guides/tree/master/data
+
+When using Velveteen in particular, try to follow:
+
+* Keep workers focused on a single task and interact with at most one service
+  (e.g. S3, Postgres, third-party API).
+* Name workers after the actions they perform. E.g. a worker that fetches a
+  user's most recent GitHub commits could be called `FetchUserCommits`.
+  Velveteen will use the same name for the worker's queue.
+* In general, routing keys should be past tense events. E.g. `FetchUserCommits`
+  could publish the API response with a routing key of `user.commits.fetched`.
+  This documents what already happened and makes no assumptions about what
+  happens next. If a worker is the first step in a pipeline, e.g. runs a
+  schedule, use present tense - `user.commits.fetch`.
```

Comment:
I got a little confused about this. How would you feel about 2 guidelines?

* For workers that are the first step in a pipeline (e.g. runs on a schedule), name routing keys in the present tense – `user.commits.fetch`
* For workers that are not the first step, routing keys should be in the past tense (e.g. `FetchUserCommits` could publish the API response with a routing key of `user.commits.fetched`. This documents what already happened and makes no assumptions about what happens next.

Reply:
I like that! Updated in 1191388118a06fc8be1d851c014c0d2f0c6e2463, expanding on the (contrived) examples.